pythonforandroid/pythonpackage.py

```diff
@@ -0,0 +1,684 @@
+            output = e.output.decode('utf-8', 'replace')
+            raise ValueError(
+                'venv creation unexpectedly ' +
+                'failed. error output: ' + str(output)
+            )
+        venv_path = os.path.join(venv_parent, "venv")
+
+        # Update pip and wheel in venv for latest feature support:
+        try:
+            filenotfounderror = FileNotFoundError
+        except NameError:  # Python 2.
+            filenotfounderror = OSError
+        try:
+            subprocess.check_output([
+                os.path.join(venv_path, "bin", "pip"),
+                "install", "-U", "pip", "wheel",
+            ])
+        except filenotfounderror:
+            raise RuntimeError(
+                "venv appears to be missing pip. "
+                "did we fail to use a proper system python??\n"
+                "system python path detected: {}\n"
+                "os.environ['PATH']: {}".format(
+                    _get_system_python_executable(),
+                    os.environ.get("PATH", "")
+                )
+            )
+
+        # Create download subfolder:
+        os.mkdir(os.path.join(venv_path, "download"))
+
+        # Write a requirements.txt with our package and download:
+        with open(os.path.join(venv_path, "requirements.txt"),
+                  "w", encoding="utf-8"
+                 ) as f:
+            def to_unicode(s):  # Needed for Python 2.
+                try:
+                    return s.decode("utf-8")
+                except AttributeError:
+                    return s
+            f.write(to_unicode(transform_dep_for_pip(dependency)))
+        try:
+            subprocess.check_output(
+                [
+                    os.path.join(venv_path, "bin", "pip"),
+                    "download", "--no-deps", "-r", "../requirements.txt",
+                    "-d", os.path.join(venv_path, "download")
+                ],
+                stderr=subprocess.STDOUT,
+                cwd=os.path.join(venv_path, "download")
+            )
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("package download failed: " + str(e.output))
+
+        if len(os.listdir(os.path.join(venv_path, "download"))) == 0:
+            # No download. This can happen if the dependency has a condition
+            # which prohibits install in our environment.
+            # (the "package ; ... conditional ... " type of condition)
+            return (None, None)
+
+        # Get the result and make sure it's an extracted directory:
+        result_folder_or_file = os.path.join(
+            venv_path, "download",
+            os.listdir(os.path.join(venv_path, "download"))[0]
+        )
+        dl_type = "source"
+        if not os.path.isdir(result_folder_or_file):
+            # Must be an archive.
+            if result_folder_or_file.endswith(".zip") or \
+                    result_folder_or_file.endswith(".whl"):
```

Comment:
Minor: you can pass a tuple and skip the `or`
```python
result_folder_or_file.endswith((".zip", ".whl"))
```

Reply:
done!